client/lib/pages/symptom_checker/symptom_checker_view.dart

```diff
@@ -0,0 +1,199 @@
+    if (_model != null) {
+      return;
+    }
+    // TODO: Reduce this boilerplate
+    //Locale locale = Localizations.localeOf(context);
+    try {
+      final countryIsoCode =
+          (await UserPreferencesStore.readFromSharedPreferences())
+              .countryIsoCode;
+      final logicContext =
+          await LogicContext.generate(isoCountryCode: countryIsoCode);
+      final store = widget.dataSource;
+      await store.update();
+      var content = store.symptomChecker;
+      await Dialogs.showUpgradeDialogIfNeededFor(context, content);
+      _model = SymptomCheckerModel(content, logicContext);
+    } catch (err, stacktrace) {
+      print('Error loading content: $err\n$stacktrace');
+    }
+    _model.addListener(_modelChanged);
+    if (!mounted) return;
+    setState(() {});
+  }
+
+  Future<bool> _onWillPop() async {
+    // If the page controller hasn't been built yet then we're on the first page
+    if (_controller.hasClients == false) return true;
+
+    if (inTransition) return false;
+
+    if (_controller.page == 0) {
+      return true;
+    } else {
+      await goBack();
+      return false;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: _onWillPop,
+      child: Material(
+        color: Constants.backgroundColor,
+        child: Column(
+          children: <Widget>[
+            PageHeader(
+              inSliver: false,
+              title: 'Check-Up',
+              appBarColor: Constants.backgroundColor,
+              trailing: FlatButton(
+                padding: EdgeInsets.zero,
+                child: ThemedText(
+                  'Cancel',
+                  variant: TypographyVariant.button,
+                  style: TextStyle(color: Constants.whoBackgroundBlueColor),
+                ),
+                onPressed: () =>
+                    Navigator.of(context).popUntil((route) => route.isFirst),
+              ),
+            ),
+            Expanded(child: _buildPage(context)),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildPage(BuildContext context) {
+    if (_model == null) {
+      return _buildMessage('Loading...', loading: true);
+    }
+    if (_model.isFatalError) {
+      return _buildMessage(
+          'Unfortunately the symptom checker encountered an error.  If this is an emergency, please call for help immediately.',
+          loading: false);
+    }
+    return PageView(
+        physics: NeverScrollableScrollPhysics(),
+        controller: _controller,
+        children: _pages ?? []);
+  }
+
+  Widget _viewForPageModel(SymptomCheckerPageModel model) {
+    switch (model.question.type) {
+      case SymptomCheckerQuestionType.SingleSelection:
+        return ShortListQuestionView(pageDelegate: this, pageModel: model);
+      case SymptomCheckerQuestionType.MultipleSelection:
+        return ShortListQuestionView(pageDelegate: this, pageModel: model);
+    }
```

Comment:
FirebaseAnalytics should log here too.... this might be caught by the top level main method:
https://github.com/WorldHealthOrganization/app/blob/5d4f17f49cbcfbea0c2badbdff0e35d6f278eb36/client/lib/main.dart#L83-L94

Reply:
Maybe these should be the same error with different parameters?  I don't want to introduce more complexity to the event logs just for this so I'm going to call this `SymptomCheckerModelError2` for now.

Reply:
Error2 is fine for now, main thing is to know an error is happening